### ISSUE TITLE
[Request] Engine & Propulsion related outfits and mods

### ISSUE THREAD

**Issue by Zitchas** [endless-sky/Endless-Sky-Delta#118]
### Problem Description

Historically, any excess engine space people have available has virtually always been used up by simply buying more engines. Now that we have a slot system where most ships will only be able to mount a single thruster, a single steering, and a single reverse thruster; unused space is going to be more common. One the things that I was hoping for in the creation of this situation is more space for diversity in terms of both engines and engine related outfits. 

Engines:
- Since engines are no longer stackable, bigger engines have one major, unavoidable and usually dominant advantage over any smaller engines in the same category: More total thrust. This should generally always be the case. Everything else, though, is now much more open to being shaken up: For example, in the past each bigger engine absolutely must have had higher thrust-per-space because if a smaller engine had better thrust for space, then the bigger engine would never be used, and people would just fill the space with the highest density thrust-per-space engines they could find. Now it becomes a trade-off: Do you want the absolute most thrust possible? Or do you want the most *efficient* thrust-per-space? Picking the later will get you a more efficient ship with more space to spend on other things, but won't be as fast since the larger engine has more total thrust. Note: This change hasn't been implemented, it is just something that can be done.
- In short, I look forward to seeing what ideas people can come up with for making engines more unique engines are possible when they don't have to comply so strictly to the "next size bigger must be both more thrust and more thrust-per-space than the smaller one."

Accessories:
- This, to be honest, is where I really would like to see a lot of innovation happen. I am also 100% in favor of seeing more code implement to support more innovation. I would very much like to see some percentage based modifiers here.
- What engine-related things could we equip on a spaceship to change how it handles?

ideas:
- fuel purifier that decreases fuel consumption by 10% (everything that uses fuel, from flamethrowers to afterburners to hyperspace drives)
- engine self-cleaning system that decreases heat production of the engines by 10%
- engine auto-tuning system that takes 10% of the available engine space on the ship, and gives a +10% thrust and +10% steering boost
- Engineer's hammock: Gives +1 bunk on your ship but only takes up engine space, not outfit space.
- Hydronic heating loop: Heats the living area of the ship with heat pulled directly off the engine. Reduce heat output of the engine by 10 per bunk on the ship, but takes up 2 space per bunk on the ship. (cannot reduce heat below the total heat produced by the engines)
- Reversed afterburner projection scoop: Uses the afterburner slot to generate a giant field to collect fuel, but also acts as a brake to slow the ship down (not a reverse thruster, won't go backwards, just slows the ship)
- Inertial dampner: Reduces the mass of the ship by 10%.
- Non-inertial navigation: affectionately known to some as tractor propulsion. The ship only moves when you intentionally apply a force. As soon as a force is not applied, the ship comes to a halt instantly. 

I am sure there are many more such ideas of things that could be related to propulsion. I look forward to hearing them!

### Related Issue Links

I'm sure there are some, but don't have time to find them right now.

### Desired Solution

Lots of them. All of the above. Variety is really the spice of the game, and the more options the better. Ideally, these should contribute towards making each faction a bit more unique and flavorful. 

### Alternative Approaches

n/a

**Comment by TheGiraffe3** [comment]
The code for many of these already exists in-game (and actually the Successors have something like the reversed afterburner projection scoop) but I imagine the hardest problem will be sprites for new outfits.
There isn't really anybody who does artwork helping with Delta, right?

**Comment by Zitchas** [comment]
No, we don't currently have any artists that are explicitly helping with Delta; although to date it's been about 50% code and 50% changing data values, with "new things" being just a few scanners comparatively recently. 

The code for some of those things is there, but I don't think it is there for any of the percentage based things.

I'm definitely interested in taking a look at the successors reverse afterburner ramscoop thing, but I would be surprised if it was acting like a brake and not a reverse thruster. I didn't think we had code for purely braking.

In any case, ideas is what this issue is for (as well as to hopefully attract people interested in implementing any of them)

Just to be clear, the above ideas I sort of expect to be spread out across many factions. Some things will probably exist in many forms. Like fuel purifier and other engine tweaks. I wouldn't be surprised if most factions had something like that. Any faction that encourages invention, tweaking, and modding their own stuff would. Factions that are rigid, non-individualistic, don't promote non-standardized systems, etc would not. Things like inertial dampeners or tractor propulsion would likely be advanced factions only.
